File: Scala/code/src/main/scala/LogicalProblems/Basic/Factorial.scala
package LogicalProblems.Basic

import scala.io.StdIn

object Factorial extends App {
  println("enter a number")
  val input = StdIn.readInt()
  val factorial = getFactorial(input)
  println(factorial)

  // without recursion
  println((1 to input).reduce((a,b) => a * b))

  println((1 to input).product)

  // recursive
  private def getFactorial(num: Int): Int ={
    if(num>1)
      num * getFactorial(num-1)
    else
      1
  }
}

File: python/code/src/basicProblems/Factorial.py
"""
    Program to calculate the factorial of an input number:
"""

def getFactorial(num:int):
    if num == 1:
        return 1
    else:
        return num * getFactorial(num - 1)

if __name__ == '__main__':
    num = int(input("Enter a number: "))
    factorial = getFactorial(num)
    print(factorial)